src/display/display_utils.js

```diff
@@ -635,6 +635,59 @@ class PDFDateString {
   }
 }
 
+function makeColorComp(n) {
+  return Math.floor(Math.max(0, Math.min(1, n)) * 255)
+    .toString(16)
+    .padStart(2, "0");
+}
+
+const ColorConverters = {
+  // PDF specifications section 10.3
```

Comment:
I don't think it matters that much for how this will be used, but we do have a bunch of colorspace conversion code that does a better job than the simplistic conversions given by the spec: https://github.com/mozilla/pdf.js/blob/master/src/core/colorspace.js

Reply:
I saw that but afaik the end goal is to be able to set a color in a textfield or whatever I think too that it doesn't matter.
FYI, pdfium code does the same here.